src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodCapiDriver : GpioDriver
+    {
+        private const string _basePath = "/dev/";
+
+        public string GpioChip { get; set; } = "gpiochip0";
+
+        private SafeChipHandle _chip;
+
+        private Dictionary<int, SafeLineHandle> _pinNumberToSafeLineHandle;
+
+        private timespec _defaultTimeOut = new timespec(IntPtr.Zero, 1_000_000); // one millisecond
+
+        private Dictionary<int, LibgpiodDriverEventHandler> _pinNumberToEventHandler = new Dictionary<int, LibgpiodDriverEventHandler>();
+
+        protected internal override int PinCount
+        {
+            get
+            {
+                return (int)Interop.gpiod_chip_num_lines(_chip);
+            }
+        }
+
+        public LibgpiodCapiDriver()
+        {
+            OpenChip();
+            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+        }
+
+        public LibgpiodCapiDriver(string chip)
+        {
+            GpioChip = chip;
```

Comment:
>  GpioChip = chip; [](start = 10, length = 18)

do we need to test for null value?

Reply:
for me not needed, do you think we need?

Reply:
what happen when the user passing null? we'll throw in OpenChip? if so then we should be fine and not needed to checj for null.

---
In reply to: [244898830](https://github.com/dotnet/iot/pull/133#discussion_r244898830) [](ancestors = 244898830)

Reply:
go Segmentation fault error, will add null check